### ISSUE TITLE
Release tests v4.21

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/action#3557]
Add the `release` label to this issue
## The basics
- [x] Smoke tested the demo, unauthenticated
- [x] Created an account
- [x] Verified invite via mass link works
- [x] Verified invite via email works
- [x] Added Slack, verified meeting notification
- [x] Added GitHub, verified issue created
- [x] Added Jira, verified issue created
- [x] Smoke tested the retro meeting with 2 players
- [x] Smoke tested the check-in meeting with 2 players
- [x] Smoke tested cards on the dashboard
- [x] Created a 2nd team
- [x] Upgraded to Pro
## What’s changed
Complete the tests for all merged PRs:
- List each PR with a header and link to ([PR title], #[issue id])
- Copy and paste tests into this issue
- Run the equivalent test on staging that you would run in a local environment as closely as possible (e.g. updating the database to simulate cases, etc.)
- Add additional tests not covered in PRs as needed
#### Meeting menu, real-time updates
- [x] User 1 & User 2 on same team. User 1 in a meeting, User 2 on the dash.
- [x] User 2 looks at active meeting menu, User 1 changes meeting name, User 2 looks at active meeting menu, sees name change
- [x] User 1 advances the meeting, User 2 sees the updated phase
 
#### Mobile header
- [x] browser around the app in mobile view. the header has a good title (e.g. My Timeline, My Tasks, team name, etc)

#### Bouncing next button
- [x] go to reflect phase, wait 2 mins, see the button bounce. then add a timer for 1 min. it doesn't bounce until the timer runs out.
- [x] go to group phase, wait 1 min, it bounces. add timer for 1 min, bounces after 1 min. group a card, it stops bouncing for 30 seconds.

#### Demo
- [x] do the demo, the robot avatars appear in the upper right
- [x] add a timer, it works
- [x] the grouping animations are a lot less janky
- [x] group a couple cards, press escape to collapse. it works

### Require verified emails for blacklisted domains #3541 
- [x] run the migration, enable mailgun API key so emails get sent
- [x] add a blacklisted domain to the database, e.g. `r.db('actionDevelopment').table('SecureDomain').insert({domain: 'parabol.co'})`
- [x] copy a mass invitation link
- [x] paste the mass invite link in an incognito window, when it asks you to sign up, sign up using an email that you control that is blacklisted, e.g. `matt+foobar@parabol.co`
- [x] see the error says you need to check your email
- [x] check your email & see a link
- [x] click the link. it creates your account & accepts the team invitation
- [x] close out, revisit the link, it serves as a login link
- [x] expire the token `r.db('actionDevelopment').table('EmailVerification').filter({email: 'matt+foobar@parabol.co'}).update({expiration: new Date()})`
- [x] go to the link again, it tells you it's expired 
- [x] go to the link with a bad token in it, it tells you it's invalid

### Fix connectivity #3543 
- [x] go to a team dash, then quit the server, the disconnected snackbar appears. start the server up again, the toast goes away
- [x] to to a team dash, get the ID of one of the tasks, then run a query that noticeably updates the task, e.g. `r.db('actionDevelopment').table('Task')
  .get('NpnJoWeo')
  .update({status: 'stuck'})`. quit the server, then start the server again. notice that when the client reconnects, the status updates to stuck

#### Also fixes the small Task editing status bug
- [x] click a task, it says editing. set the due date, it doesn't say editing

### Test Group DnD
- See the reproduction steps from #3388 
- [x] Group DnD works with 2 players

**Comment by ackernaut** [comment]
## Test cycle no. 1 notes WIP
### DnD issues (e.g. in demo, unauthenticated)
- Try grouping and ungrouping some reflections

![demo-v4 21](https://user-images.githubusercontent.com/307286/73859722-1a7fbb80-4800-11ea-9cda-2d710aec78df.gif)
### Participant out of sync
- Start a retro with 2 players
- The facilitator advances through the social check-in, but the participant gets left behind, and the rejoin button appears

**Comment by mattkrick** [comment]
timer flickers when grouping across columns, there some borked refresh logic here

**Comment by ackernaut** [comment]
### Participant out of sync
- Start a retro with 2 players
- The facilitator advances through the social check-in, but the participant gets left behind, and the rejoin button appears
- This happens in other phases and stages, too
- This happens in both the Retro and Action meetings

**Comment by ackernaut** [comment]
#### New team form, mobile header
- When navigating to the new team form on mobile, the mobile header keeps the title of the last space visited (e.g. My Tasks or [Team Name])
- Other views work as expected: Org name, Organizations, Notifications, Timeline, My Tasks, [Team], My Profile

**Comment by ackernaut** [comment]
### Bouncing button in group
- This works with the timer (it stops bouncing when the timer is set; it bounces again once the timer is done)
- This works when dragging a card (it stops bouncing for 30 seconds once you drag a card)
- It doesn’t reset the bouncing after the timer has stopped, and then you drag a card (nuanced!)
  - The button bounces
  - Set the timer, it doesn’t bounce again until the timer stops
  - Drag a card, the button doesn’t stop bouncing
- Low priority

**Comment by ackernaut** [comment]
### Connectivity
- To test this I disconnected from wifi on my laptop, and updated a card on the team dash using my phone
- At some point the dash had the offline snackbar on the laptop. When it reconnected the card was updated
- I suppose this verifies #3543

**Comment by mattkrick** [comment]
> out of sync

fixed

> team form header

fixed

> flickering clock

fixed

> bad DnD

???

**Comment by ackernaut** [comment]
It seems that the fix for the flickering timer also fixed the DnD issue as the facilitator was the one who saw both of those issue with or after a running timer.

**Comment by ackernaut** [comment]
Verified fixes:
- Participant out of sync
- Team form mobile header
- Flickering timer
- Bad DnD

**Comment by mattkrick** [comment]
bad DnD was likely caused by the relay bug thinking components are suspending.
it's a bug in relay, mitigated it by setting one of the relay internal feature flags.

**Comment by mattkrick** [comment]
OK, real cause was a `touches` array of touch events was sent with the mouse event... SOMETIMES. 
so, instead of trusting the touches array, we now just trust the `type` fields, which is a little more verbose, but safer.

**Comment by ackernaut** [comment]
### Dragging out of the group dialog
- Group multiple cards
- Drag a card out of the dialog, making sure the mouse travels outside of the dialog, and see the card ungroup as intended
- Drag a card out of the dialog, but instead of traveling outside of the dialog, drop before you get outside of the dialog. The card animates like it’s ungrouping, but the real card stays in the dialog
- I don’t know if this has anything to with the z-index

![demo-v4 21-dragoutofgroup](https://user-images.githubusercontent.com/307286/73873638-8faabb00-4817-11ea-93ad-aeb9ee0d84c7.gif)

**Comment by ackernaut** [comment]
### Demo: add card during group
- Go to the demo, unauthenticated
- Advance to the Group phase
- Try adding a new card using the `add` control
- See an error

![Screenshot 2020-02-05 13 45 41](https://user-images.githubusercontent.com/307286/73877262-f337e700-481d-11ea-9120-c70bacd5fce3.png)
